extensions/obsidian/src/openvault.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  List,
+  ActionPanel,
+  OpenAction,
+  getPreferenceValues,
+  showToast,
+  ToastStyle
+} from "@raycast/api";
+import { useEffect, useState } from "react";
+
+interface Preferences {
+  vaults: String;
+}
+
+interface Vault {
+  name: string;
+  key: number;
+}
+
+function parseVaults() {
+  const pref: Preferences = getPreferenceValues();
+  const vaultstring = pref.vaults
+  let vaults = [];
+
+  if (vaultstring) {
+    const strings = vaultstring.split(",");
+    for (let i = 0; i < strings.length; i++) {
+      vaults.push({
+        name: strings[i].trim(),
+        key: i
```

Comment:
similar comment about empty vault names. It's also not recommended to use an index as a react key. Is it possible to have vaults with the same name? otherwise I'd recommend to use the name as the key

Reply:
No, a vault name should be unique. I am now using the name as key.